File: frontend/src/app/model/Match.ts
import { Event } from './Event';

export class Match {
    id: number;
    home: string;
    away: string;
    isActive: boolean;
    homeScore: number;
    awayScore: number;
    htHomeScore: number;
    htAwayScore: number;
    events: Array<Event>;

    constructor(home?: string, away?: string, isActive?: boolean, homeScore?: number, awayScore?: number, htHomeScore?: number, htAwayScore?: number, events?: Array<Event>, id?: number) {
        this.home = home;
        this.away = away;
        this.isActive = isActive;
        this.homeScore = homeScore;
        this.awayScore = awayScore;
        this.htHomeScore = htHomeScore;
        this.htAwayScore = htAwayScore;
        this.events = events;
        this.id = id;
    }
}

File: src/main/java/hu/elte/easybet/model/Match.java
package hu.elte.easybet.model;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name = "matches")
@Data
@AllArgsConstructor
@NoArgsConstructor
//@EqualsAndHashCode(callSuper = true)
public class Match extends BaseEntity{

    @Column(nullable = false)
    private String home;
    
    @Column(nullable = false)
    private String away;
    
    @Column(nullable = false)
    private Boolean isActive;

    @Column
    private int homeScore;
    
    @Column
    private int awayScore;
    
    @Column
    private int htHomeScore;
    
    @Column
    private int htAwayScore;

    @JsonIgnore
    @OneToMany(mappedBy="match", cascade = CascadeType.ALL)
    private List<Event> events;
    
}
